source/features/clone-branch.tsx

```diff
@@ -0,0 +1,100 @@
+import React from 'dom-chef';
+import select from 'select-dom';
+import gitBranch from 'octicon/git-branch.svg';
+import insertTextTextarea from 'insert-text-textarea';
+import delegate, {DelegateEvent} from 'delegate-it';
+import * as api from '../libs/api';
+import features from '../libs/features';
+import loadingIcon from '../libs/icon-loading';
+import {getRepoURL, getRepoGQL} from '../libs/utils';
+
+const getBranchBaseSha = async (branchName: string): Promise<string> => {
+	const {repository} = await api.v4(`
+		repository(${getRepoGQL()}) {
+			ref(qualifiedName: "${branchName}") {
+				target {
+					oid
+				}
+			}
+		}
+	`);
+
+	return repository.ref.target.oid;
+};
+
+async function createBranch(newBranchName: string, baseSha: string): Promise<true | string> {
+	const response = await api.v3(`repos/${getRepoURL()}/git/refs`, {
+		method: 'POST',
+		body: {
+			sha: baseSha,
+			ref: 'refs/heads/' + newBranchName
+		},
+		ignoreHTTPStatus: true
+	});
+
+	return response.ok || response.message;
+}
+
+async function cloneBranch(event: DelegateEvent<MouseEvent, HTMLButtonElement>): Promise<void> {
+	const cloneButton = event.delegateTarget;
+	const branchElement = cloneButton.closest<HTMLElement>('[data-branch-name]')!;
+
+	const currentBranch = getBranchBaseSha(branchElement.dataset.branchName!);
+	let newBranchName = prompt('Enter the new branch name')?.trim();
+
+	const spinner = loadingIcon();
+	spinner.classList.add('mr-2');
+	while (newBranchName) {
+		cloneButton.replaceWith(spinner);
+		// eslint-disable-next-line no-await-in-loop
+		const result = await createBranch(newBranchName, await currentBranch);
+		spinner.replaceWith(cloneButton);
+
+		if (result === true) {
+			break;
+		}
+
+		newBranchName = prompt(result + '\n Enter the new branch name', newBranchName)?.trim();
```

Comment:
The error doesn't look great:

<img width="312" alt="error" src="https://user-images.githubusercontent.com/1402241/77261586-d6e9ef80-6c8f-11ea-9602-5c8af9d65b0d.png">

Maybe this?

```suggestion
		newBranchName = prompt(result.replace('heads/refs/' + newBranchName, 'This') + '\nEnter the new branch name', newBranchName)?.trim();
```

Reply:
Yup. Should have done that on my own.